Assets/Scripts/Player/PlayerController.cs

```diff
@@ -106,26 +109,18 @@ public class PlayerController : MonoBehaviour {
 		else if (doAirJump)
 			Jump(airJumpCurve, airJumpEndTime);
 
-		if (Mathf.Abs(rb2D.velocity.y) < 0.01f) {
-			doJump = false;
-			doAirJump = false;
-		}
-
 		if (!doJump || !doAirJump)
 			rb2D.velocity = new Vector2(velocity.x, rb2D.velocity.y + gravity * Time.deltaTime);
 	}
 
-	private bool CheckIfGrounded() {
-		Vector2 colliderPosition = (Vector2)transform.position + playerCollider.offset;
-		Vector2 rayStartPosition = new Vector2(colliderPosition.x, colliderPosition.y - playerCollider.bounds.extents.y);
-
-		RaycastHit2D hit = Physics2D.Raycast(rayStartPosition,Vector2.down, groundedDistance, groundLayer);
-
-	#if UNITY_EDITOR
-		Debug.DrawRay(rayStartPosition, Vector3.down * groundedDistance, Color.red);
-	#endif
+	private void OnTriggerEnter2D(Collider2D other) {
+		if (other.CompareTag("Ground"))
+			isGrounded = true;
+	}
 
-		return hit.collider != null;
+	private void OnTriggerExit2D(Collider2D other) {
+		if (other.CompareTag("Ground"))
+			isGrounded = false;
```

Comment:
This logic is flawed. If we were to touch two ground colliders we get two `OnTriggerEnter2D` events, if we then move so that we are only touching one of them we get an `OnTriggerExit2D` event which sets `isGrounded` to false even though we are still touching a ground collider.

Reply:
Also, using tags for the ground check is a bad idea.

Imagine we have a box with the tag `Box`. Now we want to be able to jump while standing on that box, so we need it to have the `Ground` tag. But an object can only have one tag!

I would stick with layers and use something like this instead:
```cs
if (groundCheckCollider.IsTouchingLayers(groundLayer))
    isGrounded = true;
```

Reply:
True with the logic, I had some trouble with checking layers using built on trigger enter/exit so will take an extra look.